### PR TITLE
feat(stop_filter): componentize StopFilter

### PR DESCRIPTION
**Pull request by a-maumau**
## Description

Componentize the node in localization/stop_filter.
(Add glog feature to localization/StopFilter)

Log output is changed from "screen" to "both"
Add glog feature to localization/StopFilter.

## Tests performed

confirmed that killing the ekf_localizer process will output logs to terminal and log file.
``` text
1716456916.0847893 [component_container_mt-57] [INFO] [1716456916.084236188] [control.trajectory_follower.lane_departure_checker_node]: waiting for route msg...
1716456918.1749604 [stop_filter_node-42] [INFO] [1716456918.174565969] [rclcpp]: signal_handler(signum=15)
1716456918.1751084 [stop_filter_node-42] *** Aborted at 1716456918 (unix time) try "date -d @1716456918" if you are using GNU date ***
1716456918.1757030 [stop_filter_node-42] PC: @                0x0 (unknown)
1716456918.1757896 [stop_filter_node-42] *** SIGTERM (@0x3e8001c1f43) received by PID 1192177 (TID 0x7f35ada78680) from PID 1843011; stack trace: ***
1716456918.1762626 [stop_filter_node-42]     @     0x7f35ae3164d6 google::(anonymous namespace)::FailureSignalHandler()
1716456918.1766379 [stop_filter_node-42]     @     0x7f35ae219ded rclcpp::SignalHandler::signal_handler()
1716456918.1771111 [stop_filter_node-42]     @     0x7f35ad842520 (unknown)
1716456918.1771710 [stop_filter_node-42]     @     0x7f35ad891117 (unknown)
1716456918.1772175 [stop_filter_node-42]     @     0x7f35ad893a41 pthread_cond_wait
1716456918.1772585 [stop_filter_node-42]     @     0x7f35ad716a8d ddsrt_cond_wait
1716456918.1776760 [stop_filter_node-42]     @     0x7f35ad716b55 ddsrt_cond_waituntil
1716456918.1777370 [stop_filter_node-42]     @     0x7f35ad6fd45d (unknown)
1716456918.1777802 [stop_filter_node-42]     @     0x7f35ad7b974f rmw_wait
1716456918.1781502 [stop_filter_node-42]     @     0x7f35adf57848 rcl_wait
1716456918.1784594 [stop_filter_node-42]     @     0x7f35ae1611b6 rclcpp::Executor::wait_for_work()
1716456918.1788106 [stop_filter_node-42]     @     0x7f35ae1616d3 rclcpp::Executor::get_next_executable()
1716456918.1791706 [stop_filter_node-42]     @     0x7f35ae165c81 rclcpp::executors::SingleThreadedExecutor::spin()
1716456918.1792378 [stop_filter_node-42]     @     0x562f91c6f486 main
1716456918.1795518 [stop_filter_node-42]     @     0x7f35ad829d90 (unknown)
1716456918.1796126 [stop_filter_node-42]     @     0x7f35ad829e40 __libc_start_main
1716456918.1796575 [stop_filter_node-42]     @     0x562f91c6fe35 _start
1716456918.1809032 [ERROR] [stop_filter_node-42]: process has died ...
```

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by SakodaShintaro — APPROVED** [review]
Thank you very much!
I have confirmed that the logging_simulator works properly and that glog works properly when using the [tier4 version of rclcpp](https://github.com/tier4/rclcpp).

File: localization/stop_filter/launch/stop_filter.launch.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <arg name="input_odom_name" default="ekf_odom"/>
   <arg name="output_odom_name" default="stop_filter_odom"/>
   <arg name="debug_stop_flag" default="debug/stop_flag"/>
-  <node pkg="stop_filter" exec="stop_filter" name="stop_filter" output="screen">
+  <node pkg="stop_filter" exec="stop_filter_node" output="screen">
```
Comment (SakodaShintaro) [inline localization/stop_filter/launch/stop_filter.launch.xml:6]: Sorry, we have decided to change the log output destination from screen to both, so please make the change. :pray:

**Review by SakodaShintaro — CHANGES_REQUESTED** [review]

File: localization/stop_filter/launch/stop_filter.launch.xml (hunk 1 above)
Comment (a-maumau) [reply localization/stop_filter/launch/stop_filter.launch.xml]: I have changed the log output from "screen" to "both" in https://github.com/autowarefoundation/autoware.universe/pull/7099/commits/4c41c2dd08351dbc2d269bfa65ec4e208afd84a9

Comment (SakodaShintaro) [reply localization/stop_filter/launch/stop_filter.launch.xml]: I checked again and confirmed that the errors were properly logged to the file.
Thanks!

**Review by SakodaShintaro — APPROVED** [review]
Looks Good To Me

**Pull request merged**